src/coreclr/jit/codegenarm64.cpp

```diff
@@ -10066,44 +10066,21 @@ void CodeGen::instGen_MemoryBarrier(BarrierKind barrierKind)
 // genCodeForMadd: Emit a madd (Multiply-Add) instruction
 //
 // Arguments:
-//     tree - GT_MADD tree where op1 or op2 is GT_ADD
+//     tree - GT_MADD tree where op2 is GT_MUL
 //
 void CodeGen::genCodeForMadd(GenTreeOp* tree)
 {
     assert(tree->OperIs(GT_MADD) && varTypeIsIntegral(tree) && !(tree->gtFlags & GTF_SET_FLAGS));
     genConsumeOperands(tree);
 
-    GenTree* a;
-    GenTree* b;
-    GenTree* c;
-    if (tree->gtGetOp1()->OperIs(GT_MUL) && tree->gtGetOp1()->isContained())
-    {
-        a = tree->gtGetOp1()->gtGetOp1();
-        b = tree->gtGetOp1()->gtGetOp2();
-        c = tree->gtGetOp2();
-    }
-    else
-    {
-        assert(tree->gtGetOp2()->OperIs(GT_MUL) && tree->gtGetOp2()->isContained());
-        a = tree->gtGetOp2()->gtGetOp1();
-        b = tree->gtGetOp2()->gtGetOp2();
-        c = tree->gtGetOp1();
-    }
-
-    bool useMsub = false;
-    if (a->OperIs(GT_NEG) && a->isContained())
-    {
-        a       = a->gtGetOp1();
-        useMsub = true;
-    }
-    if (b->OperIs(GT_NEG) && b->isContained())
-    {
-        b       = b->gtGetOp1();
-        useMsub = !useMsub; // it's either "a * -b" or "-a * -b" which is the same as "a * b"
-    }
+    GenTree* a = tree->gtGetOp1();
+    GenTree* b = tree->gtGetOp2()->gtGetOp1();
```

Comment:
Is `tree->gtGetOp2()` expected to be GT_MUL? Is that because we don't have a GenTree node that contains all three arguments directly (should we? like some of the block ops, for example) Can you assert that?